monkey/infection_monkey/ransomware/ransomware_payload.py

```diff
@@ -44,12 +47,16 @@ def _encrypt_files(self, file_list: List[Path]) -> List[Tuple[Path, Optional[Exc
             try:
                 self._encryptor.encrypt_file_in_place(filepath)
                 self._add_extension(filepath)
-                results.append((filepath, None))
+                self._send_telemetry(filepath, "")
             except Exception as ex:
-                results.append((filepath, ex))
+                self._send_telemetry(filepath, str(ex))
 
         return results
 
     def _add_extension(self, filepath: Path):
         new_filepath = filepath.with_suffix(f"{filepath.suffix}{self._new_file_extension}")
         filepath.rename(new_filepath)
+
+    def _send_telemetry(self, filepath: Path, error: str):
+        encryption_attempt = RansomwareTelem((str(filepath), str(error)))
+        self._telemetry_messenger.send_telemetry(encryption_attempt)
```

Comment:
Why can't it be `RansomwareTelem((str(filepath), str(error))).send()` or `RansomwareTelem.send((str(filepath), str(error))` ?

Reply:
See response to https://github.com/guardicore/monkey/pull/1259/files#r658519175